src/app/(main)/my-voyage/[teamId]/sprints/components/forms/AgendaTopicForm.tsx

```diff
@@ -0,0 +1,343 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useParams, useRouter } from "next/navigation";
+import { SubmitHandler, useForm } from "react-hook-form";
+import * as z from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { TrashIcon } from "@heroicons/react/20/solid";
+
+import Button from "@/components/Button";
+import TextInput from "@/components/inputs/TextInput";
+import Textarea from "@/components/inputs/Textarea";
+
+import { validateTextInput } from "@/helpers/form/validateInput";
+import { useSprint, useAppDispatch } from "@/store/hooks";
+import { Agenda } from "@/store/features/sprint/sprintSlice";
+import useServerAction from "@/hooks/useServerAction";
+import {
+  addAgendaTopic,
+  editAgendaTopic,
+} from "@/myVoyage/sprints/sprintsService";
+import { onOpenModal } from "@/store/features/modal/modalSlice";
+import routePaths from "@/utils/routePaths";
+import Spinner from "@/components/Spinner";
+import { persistor } from "@/store/store";
+
+const validationSchema = z.object({
+  title: validateTextInput({
+    inputName: "Title",
+    required: true,
+  }),
+  description: validateTextInput({
+    inputName: "Description",
+    required: true,
+  }),
+});
+
+export type ValidationSchema = z.infer<typeof validationSchema>;
+
+export default function AgendaTopicForm() {
+  const [isMounted, setIsMounted] = useState(false);
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
```

Comment:
What is this for?

Reply:
Haha 😀 I had a problem with a textarea field again (height didn't want to adjust when a default value is passed) so this `isMounted` trick seems to solve it... but I've just looked at it again, seems like I forgot to pass default values at all 😅